libraries/pybamm-developer/01_ode.md

```diff
@@ -0,0 +1,341 @@
+later once we have defined the expressions for the ODEs.
+
+## PyBaMM parameters
+
+The reservoir model has a number of parameters that need to be defined. In PyBaMM a parameter can be defined using the [`pybamm.Parameter`](https://docs.pybamm.org/en/stable/source/api/expression_tree/parameter.html#parameter) class. For example, if you wanted to define a parameter with name "a", you would write
+    
+```python
+a = pybamm.Parameter("a")
+```
+
+The name of the parameter is used to identify it in the model, the name of the Python variable you assign it to is arbitrary.
+
+You can also define a parameter that is defined as a function using the [`pybamm.FunctionParameter`](ihttps://docs.pybamm.org/en/stable/source/api/expression_tree/parameter.html#pybamm.FunctionParameter) class, which is useful for time varying parameters such as the current, or functions like the OCV function parameters $U_p(x_p)$ and $U_n(x_n)$, which are both functions of the stochiometries $x_p$ and $x_n$. For example, to define a parameter that is a function of time, you would write
+
+```python
+I = pybamm.FunctionParameter("I", {"Time [s]": pybamm.t})
+```
+
+where `pybamm.t` is a special variable that represents time.
+
+::::challenge{id="ode-parameters" title="Define the parameters for the reservoir model"}
+
+Define the parameters for the reservoir model, including the current $I(t)$, the OCV functions $U_p(x_p)$ and $U_n(x_n)$, the capacities $Q_n$ and $Q_p$, and the resistance $R$.
+
+:::solution
+```python
+i = pybamm.FunctionParameter("Current function [A]", {"Time [s]": pybamm.t})
+x_n_0 = pybamm.Parameter("Initial negative electrode stochiometry")
+x_p_0 = pybamm.Parameter("Initial positive electrode stochiometry")
+U_p = pybamm.FunctionParameter("Positive electrode OCV", {"x_p": x_p})
+U_n = pybamm.FunctionParameter("Negative electrode OCV", {"x_n": x_n})
+Q_n = pybamm.Parameter("Negative electrode capacity [A.h]")
+Q_p = pybamm.Parameter("Positive electrode capacity [A.h]")
+R = pybamm.Parameter("Electrode resistance [Ohm]")
+```
+:::
+::::
+
+## A PyBaMM Model
+
+Now that we have defined the parameters and variables for the reservoir model,
+we can define the model itself. In PyBaMM a model can be defined using the
+[`pybamm.BaseModel`](https://docs.pybamm.org/en/stable/source/api/models/base_models/base_model.html)
+class. For example, if you wanted to define a model with name "my model", you
+would write
+
+```python
+model = pybamm.BaseModel("my model")
+```
+
+This class has four useful attributes for defining a model, which are:
+1. `rhs` - a python dictionary of the right-hand-side equations
+2. `algebraic` - a python dictionary of the algebraic equations (we won't need this for our ODE model)
+3. `initial_conditions` - a python dictionary of the initial conditions
+4. `variables` - a python dictionary of the output variables
+
+As an example, lets define a simple model for exponential decay with a single state variable $x$ and a single parameter $a$:
```

Comment:
Can we add latex of the model here, just to be very clear on the correspondence between model and pybamm?